### ISSUE TITLE
[question] Is the hidden state necessary or not in the model forward function?

### ISSUE THREAD

**Issue by kirk86** [tommyip/mamba2-minimal#3]
When trying to fit the model on some random data I get an issue when passing the hidden state back to the model.
That's due to the fact that the step has the following line:

`assert u.shape[1] == 1, "Only one token can be decoded per inference step"`

for instance the following code would results in issue:
```
for epoch in range(num_epochs):
    hidden_state = None  # Initialize hidden state at the beginning of each epoch
    for input_ids, targets in data_loader:
        optimizer.zero_grad()
        
        # Forward pass
        logits, hidden_state = model(input_ids, hidden_state)
        
        # Compute loss
        loss = criterion(logits, targets)
        loss.backward()
        optimizer.step()
```

in my case the inputs are $X = [B, L]$ and labels $Y = [B, L, V]$ where $V = vocab_size$, i.e., labels are one-hot encoded.
